[scripts|cmake] use ENV{VisualStudioVersion} to switch cmake generator (#26339)
microsoft/vcpkg: docs/users/triplets.md

Also available as build-type specific `VCPKG_CMAKE_CONFIGURE_OPTIONS_DEBUG` and `VCPKG_CMAKE_CONFIGURE_OPTIONS_RELEASE` variables.

### VCPKG_CMAKE_VS_GENERATOR

> Sets the default VS generator vcpkg will use to configure cmake if a port can not use the Ninja generator. 

Reviewer: ```suggestion
Sets the default VS generator vcpkg will use to configure CMake if a port must use MSBuild.
```
